### PR TITLE
feat: [el-dialog]add draggable

### LINKED ISSUE #4809: [Feature Request] Dialog 弹窗组件添加可拖拽配置
### Existing Component
Yes

### Component Name
Dialog 对话框

### Description
弹窗组件能否添加可拖拽配置 每次都要自己重新封装一次太麻烦了

### PR DESCRIPTION
**Pull request by Alanscut**
feat #4809 
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by baboon-king** [comment]
look forward to this future

**Comment by msidolphin** [comment]
Can you add a demo for this feature?

**Comment by Alanscut** [comment]
@msidolphin
The weird thing is that it works fine in play/App.vue

https://user-images.githubusercontent.com/21104054/147030223-90567430-37d0-4b2c-86fe-ac4cb0f2ee8d.mp4

But it is abnormal in the preview website

https://preview-4885-element-plus.surge.sh/en-US/component/dialog.html#set-draggable

**Comment by Alanscut** [comment]
@msidolphin 
https://preview-4885-element-plus.surge.sh/en-US/component/dialog.html#set-draggable

File: packages/components/dialog/src/use-dialog.ts [hunk 1]
```diff
@@ -25,9 +26,70 @@ export const useDialog = (
...
+      offsetY: 0,
+    }
+  }
 
   const style = computed<CSSProperties>(() => {
-    const style: CSSProperties = {}
+    let style: CSSProperties = {}
+    if (wrapper && props.draggable) {
+      style = {
+        left: `${transform.value.offsetX}px`,
```
Comment (msidolphin) [inline packages/components/dialog/src/use-dialog.ts:89]: consider using translate3d instead left and top

**Comment by msidolphin** [comment]
image

I think only the header area is draggable and you should change the cursor style

File: packages/components/dialog/src/use-dialog.ts (hunk 1 above)
Comment (Alanscut) [reply packages/components/dialog/src/use-dialog.ts:89]: fine

**Comment by Alanscut** [comment]
> image
> 
> I think only the header area is draggable and you should change the cursor style

only the header area is draggable is ok ,
but what is meaning of  "change the cursor style",  do you mean change the style of `el-dialog__header` like change the background-color?

**Comment by btea** [comment]
> > image
> > I think only the header area is draggable and you should change the cursor style
> 
> only the header area is draggable is ok , but what is meaning of "change the cursor style", do you mean change the style of `el-dialog__header` like change the background-color?

I think he meant to set the cursor of the `el-dialog__header` to move.
image

**Comment by Alanscut** [comment]
@btea 
ok, i got it 
it will be changed 
thank you

**Comment by Alanscut** [comment]
@msidolphin @btea 
I just checked some software in my computer and found that most of the software did not change to ` cursor:move` style when the mouse pointer was placed on the header (draggable area)
So, I don't think we need to set it.

**Comment by kooriookami** [comment]
@Alanscut It seems already have a PR #2718. I have a question. Should the dialog can be dragged out of the screen?

**Comment by Alanscut** [comment]
image
maybe drag to the top like this

**Comment by gaopeak** [comment]
Dialog 组件，怎么更改el-dialog__body 这个class 的padding，加了样式穿透，不起作用，加了custom-class 也不起作用

**Comment by gaopeak** [comment]
Dialog组件，怎么改变el-dialog__body这个class的样式，加了样式穿透，无效，加了custom-class这个，也无效

**Pull request closed without merging**